src/validators/uuid.rs

```diff
@@ -92,9 +93,11 @@ impl Validator for UuidValidator {
         let class = get_uuid_type(py)?;
         if let Some(py_input) = input.input_is_instance(class) {
             if let Some(expected_version) = self.version {
-                let py_input_version: usize = py_input.getattr(intern!(py, "version"))?.extract()?;
-                let expected_version = usize::from(expected_version);
-                if expected_version != py_input_version {
+                let py_input_version: Option<usize> = py_input.getattr(intern!(py, "version"))?.extract()?;
+                if !match py_input_version {
+                    Some(py_input_version) => py_input_version == expected_version,
+                    None => false,
+                } {
```

Comment:
```suggestion
                if !matches!(py_input_version, Some(expected_version)) {
```

Reply:
Apparently, this doesn't work. It reports "unused variable: `expected_version`" and tests fail.

Reply:
Yes, the suggestion is an equivalent to
```rust
                if !match py_input_version {
                    Some(expected_version) => true,
                    None => false,
                } {
```

which is not what we want here.

Reply:
Ah yes sorry, should have wanted:

`!matches(py_input_version, Some(v) if v == expected_version)`

or 

`!py_input_version.is_some_and(|v| v == expected_version)`